OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -356,4 +378,27 @@ private void displayCloudStatus() {
 
         LOGGER.log(message);
     }
+    
+    public FileActions fileAction(){
+        return fileActions;
+    }
+    
+    public enum CanOpenState { SAVE_AND_OPEN, OPEN, CANCEL }
```

Comment:
I have a few ideas how to rename it:
SavingWarningState,
WarningState,
UnsavedChangesState,
UnsavedChangesWarningState,
UnsavedWorkState.
If you don't like any of those, I will change it to yours "WorkInProgressState". 

Of course I can move it to another class, but it would be very small:
![small_class](https://cloud.githubusercontent.com/assets/17130832/15370164/efa9b9b8-1d35-11e6-9c95-9aa5858235d5.png)
Maybe it's a good practice, I am open for any tips.